pubsub_test.ts

```diff
@@ -79,17 +91,96 @@ test({
 });
 
 test({
-  name:
-    "testSubscriptionShouldNotThrowBadResourceErrorWhenConnectionIsClosed (#89)",
-  async fn() {
-    const redis = await connect(addr);
-    const sub = await redis.subscribe("test");
-    const subscriptionPromise = (async () => {
-      // deno-lint-ignore no-empty
-      for await (const _ of sub.receive()) {}
-    })();
-    redis.close();
-    await subscriptionPromise;
-    assert(sub.isClosed);
+  name: "testSubscribe4 (#83)",
+  // sanitizeResources: false,
+  // sanitizeOps: false,
+  async fn(): Promise<any> {
+    let parallelPromiseList: Promise<any>[] = [];
+
+    const throwawayRedisServerPort = 6464;
+    let promiseList;
+    let throwawayRedisServerChildProcess = createThrowawayRedisServer(throwawayRedisServerPort);
+
+    await delay(500);
+
+    const redisClient = await connect({ ...addr, name: "Main", port: throwawayRedisServerPort });
+    const publisherRedisClient = await connect(
+      { ...addr, maxRetryCount: 10, name: "Publisher", port: throwawayRedisServerPort }
+    );
+    const subscriberRedisClient = await redisClient.psubscribe("ps*");
+
+    const messageIterator = subscriberRedisClient.receive();
+
+    const interval = setInterval(
+      () => {
+        try {
+          parallelPromiseList.push(publisherRedisClient.publish("psub", "wayway"));
+        } catch (error) {}
+      },
+      900
+    );
+
+    setTimeout(
+      () => {
+        // console.debug("Kill the Redis server...");
+        throwawayRedisServerChildProcess.kill(Deno.Signal.SIGTERM);
```

Comment:
`Deno.Signal` is currently an unstable feature.
And I want to ensure that `deno-redis` can be used without the `--unstable` flag.
Can you change it as follows?

```suggestion
        await sendSigTerm(throwawayRedisServerChildProcess);
```

```typescript
function sendSigTerm(process: Deno.Process): Promise<void> {
  await Deno.run({
    args: ["kill", String(process.pid)],
    stdin: "null",
    stdout: "null"
  }).status();
  process.close();
}
```

Reply:
I don't know if something changed in Deno from my original tests two weeks ago, but it seems like calling `process.close()` does the job... I tried to connect client to the server after simply closing the child and it fails as expected... 🤔